libs/ts-rest/core/src/lib/client.spec.ts

```diff
@@ -691,4 +691,37 @@ describe('custom api', () => {
       })
     );
   });
+
+  it('has correct types when strict is configured', async () => {
+    const client = initClient(router, {
+      baseUrl: 'https://api.com',
+      baseHeaders: {
+        'X-Api-Key': 'foo',
+      },
+      strict: true,
+    });
+
+    fetchMock.getOnce({ url: 'https://api.com/posts' }, { status: 200 });
+
+    const result = await client.posts.getPosts({});
+
+    // @ts-expect-error 404 is not defined in the known responses
+    result.status === 404;
```

Comment:
I think you forgot this here

Reply:
This is a type level assertion that ensures the flag works as expected. If this statement no longer causes a type error it will cause an error in the type check.

Would you prefer not to have this?

Reply:
Can see it in action here https://github.com/ts-rest/ts-rest/actions/runs/5105828907/jobs/9177654229?pr=228#step:12:79

Reply:
Ohh! That's nifty! I think that pattern could be useful in some cases. In this case, I think what we generally use to assert types would be more comprehensive here and also indicate the intention of the test clearly.

```typescript
Expect<Equal<typeof result.status, 200>>
```